whisper.py

```diff
@@ -994,7 +1014,10 @@ def __archive_fetch(fh, archive, fromTime, untilTime):
   byteOrder, pointTypes = pointFormat[0], pointFormat[1:]
   points = len(seriesString) // pointSize
   seriesFormat = byteOrder + (pointTypes * points)
-  unpackedSeries = struct.unpack(seriesFormat, seriesString)
+  try:
+    unpackedSeries = struct.unpack(seriesFormat, seriesString)
+  except struct.error:
+    raise CorruptWhisperFile("Unable to read datapoints", fh.name)
```

Comment:
We have a lot of duplication here, maybe we should have `unpackSeries` and `unpackPoint` helper functions that can handle raising the exception?

Reply:
The whole part for finding the offset and reading points is duplicated four times and should be refactored. I will need a bit more time to do that. I thought this would be an improvement and would do the refactoring some other time.